app/Services/DatabaseManager/DatabaseManager.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Services\DatabaseManager;
+
+use App\Services\DatabaseManager\Exceptions\DatabaseManagerException;
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Contracts\Console\Kernel;
+use Illuminate\Process\Exceptions\ProcessFailedException;
+use Illuminate\Process\PendingProcess;
+use Symfony\Component\Console\Output\NullOutput;
+
+final class DatabaseManager implements DatabaseManagerContract
+{
+    public function __construct(
+        private Repository $config,
+        private PendingProcess $process,
+        private Kernel $artisan,
+    ) {
+    }
+
+    /**
+     * Prepare the database for transaction processing.
+     *
+     * @throws DatabaseManagerException
+     */
+    public function prepare(): void
+    {
+        $entry = sprintf('database.connections.%s.database', $this->config->get('app.env')); // @phpstan-ignore-line
+
+        is_string($file = $this->config->get($entry)) || throw DatabaseManagerException::invalidDatabaseLocation($entry);
+
+        try {
+            is_file($file) || $this->initialise($file);
+        } catch (ProcessFailedException $exception) {
+            throw DatabaseManagerException::processError($exception);
+        }
+
+        $this->artisan->call('migrate:fresh', ['--force' => true], new NullOutput());
```

Comment:
ℹ️ Calls the `migrate:fresh` command silently (thanks to `NullOutput`) and forces migrating even in production mode